extensions/amp-consent/0.1/consent-policy-manager.js

```diff
@@ -319,6 +330,36 @@ export class ConsentPolicyManager {
     });
   }
 
+  /**
+   * Wait for initial consent information to be transmitted,
+   * then get consent state for this purpose.
+   *
+   * Note: Even if we have some intiial consent info, wait until all
+   * purposes have been potentially collected, then check.
+   * @param {!Array<string>} purposes
+   * @return {!Promise<boolean>}
+   */
+  whenPurposesUnblock(purposes) {
+    return this.ConsentStateManagerPromise_.then((manager) => {
+      // Wait for all purpose consents (collected through UI or update)
+      return manager.whenHasAllPurposeConsents();
+    }).then(() => {
+      if (!this.purposeConsents_) {
+        return false;
+      }
+      let shouldUnblock = true;
+      purposes.forEach((purpose) => {
+        if (
+          !hasOwn(this.purposeConsents_, purpose) ||
+          this.purposeConsents_[purpose] === PURPOSE_CONSENT_STATE.REJECTED
+        ) {
+          shouldUnblock = false;
+        }
+      });
+      return shouldUnblock;
```

Comment:
```suggestion
      const shouldUnblock = (purpose) => !hasOwn(this.purposeConsents_, purpose) ||
          this.purposeConsents_[purpose] === PURPOSE_CONSENT_STATE.REJECTED;
      return purposes.every(shouldUnblock);
```

Reply:
Are there any performance benchmarks on using `every()` vs `forEach()`?

Reply:
`forEach` is already pretty not-great as far as loops go, and it runs for a full N loops every time without short-circuiting. `every` exits on the first falsey value (like chained `&&`) so it should be better in that respect. In general, none of these functional array methods (`forEach`/`map`/`filter`/`find` etc) will ever be as fast as a plain `for (i... ++i)` loop (which could short-circuit with early return), but since it doesn't look like this code is in critical runtime path where it's being run hundreds or thousands of times I suspect the performance difference is negligible, so `every` should be fine.